src/tools/snapshot/SnapshotRecord.java

```diff
@@ -45,26 +72,31 @@ public long getObjectPointer(final Object o) {
         "Cannot point to unserialized Objects, you are missing a serialization call: " + o);
   }
 
-  public void addMessageEntry(final long offset) {
-    this.messageOffsets.add(offset);
-  }
-
+  @TruffleBoundary
   public void addObjectEntry(final Object o, final long offset) {
     synchronized (entries) {
       entries.put(o, offset);
     }
   }
 
-  public void handleTodos(final SnapshotBuffer sb) {
+  @TruffleBoundary // TODO: convert to an approach that constructs a cache
```

Comment:
Is this TODO still current?

Reply:
only if you want to use cachedserialization nodes